tools/build_defs/repo/git.bzl

```diff
@@ -15,12 +15,84 @@
 
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "patch", "update_attrs", "workspace_and_buildfile")
 
+
+def _if_debug(cond, st, what="Action"):
+    "Print if 'cont'."
+    if cond:
+        print("{} returned {}\n{}\n----\n{}".format(what, st.return_code, st.stdout, st.stderr))
+
+def _hash(ctx, value):
+    """Hash a value... TODO needs a proper implementation.
```

Comment:
Could you also please clean up this todo?

If the TODO remains it should be made more specific:
1.  At the place of where we are calling `_hash`, it is not actually necessary. Any URL-sanitization tool would do.
2. Shelling out to bash just to compute the hash, seems wasteful and inelegant.
3. Starlark seems to provide hashing methods, and those built-in should be preferred if they can be used in this stage. (https://github.com/google/skylark/blob/master/doc/spec.md#hashing).

Otherwise, the TODO could also be just fixed and removed.

I guess this decision should be left to someone else, not me. I personally only care that my sloppy comment does not get merged like this.

Reply:
@siedentop I had more of a look at that and while the `hash` method could be used looking at the docs it does not seem to be clear if that is guaranteed to be stable across versions/implementations etc. But if we can not use the `hash` method it seems to me quite tricky to do something without shelling out as the remote naming scheme is quite quite tricky: https://stackoverflow.com/questions/41461152/which-characters-are-illegal-within-a-git-remote-name. And starlark does not provide us much to help with such a case, apart from the `replace` method.

@aehlig @laurentlb Is it safe to use the starlark built-in `hash` method to have access to a consistent hashing method? It does seem that way from the docs but is not a 100% clear.

Reply:
I'd like the `hash` function to be stable across versions/implementations, which is why it was documented here: https://docs.bazel.build/versions/master/skylark/lib/globals.html#hash

However, other implementations of Starlark don't use the same: https://github.com/bazelbuild/starlark/issues/36

Reply:
I think just having it guaranteed to be stable across all versions on Java seems fine to me. I assume if bazel ever switches out its starlark library or ever switches to another language entirely then that would probably not happen before a 2.0 and then potentially needing to clean the cache, while ideally to be avoided, should imo be acceptable.